app/scripts/modules/core/src/presentation/tables/Table.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+
+import { useIsMobile } from 'core/presentation';
+
+import { ITableRowLayoutProps } from './TableRow';
+import { ITableCellLayoutProps } from './TableCell';
+import { TableContext } from './TableContext';
+
+const { useState, useMemo } = React;
+
+export interface ITableColumn {
+  name: string;
+  sortDirection?: 'ascending' | 'descending';
+  onSort?: () => any;
+}
+
+export type ITableColumns = Readonly<Array<string | ITableColumn>>;
+
+export type ITableLayoutProps = Omit<ITableProps, 'layout' | 'columns'> & {
+  columns: ITableColumn[];
+  isMobile: boolean;
+  expandAll: () => any;
+};
+
+export interface ITableLayout {
```

Comment:
I think exporting these as components would make the most sense, given the use of React context and for composing custom layouts. 

In the case where you may need only part of the `minimalNativeTableLayout`, it feels cleaner to write:
```
<Table 
    layout={{ table: MinimumNativeTableLayout, row: CustomRow, cell:  StandardCellLayout }}
    { ...otherProps }
/>
```

Reply:
After hearing your feedback + Chris's, I am inclined to switch over to thinking about the individual layout bits as components instead of `getXYZLayout` functions. That said, I'm not 100% sure how much I want to encourage folks to do mix-and-match layout overriding. I suspect in many cases table implementations will have some implicit dependencies between — for example — the row implementation and the cell implementation.

Either way moving to think about layout implementations as components instead of functions that return elements _should_ enable the kind of overriding in your example.

Reply:
With the updates I pushed to render these layout implementation bits as components instead of functions, I believe the use case you described is now possible (though still not something I'd like to encourage often).